components/chart/index.tsx

```diff
@@ -58,10 +60,14 @@ function Chart({
     const module: NodeType = arrayOfNodes.find((module) => module.id === id);
     return module ? module : undefined;
   };
-
   const selectModule = useCallback(
     (module: NodeType, groupSelect?: boolean) => {
-      setSelectedModule(module);
+      if (selectedModuleID ===  module.id) {
```

Comment:
```suggestion
      if (selectedModule.id===  module.id) {
```
I think we can do it like that, why to create a new state variable?